### PR TITLE
quran/quran.com-frontend#224 - adds search autocomplete to the new re…

### PR DESCRIPTION
**Pull request by sharabash**
…dux code; surah matching is not working because i dont know how to get the surahs object anymore; i cant refactor request out because i dont know how to use api client.

@mmahalwy

### CODE REVIEW

File: src/scripts/components/header/SearchAutocomplete.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      this.handleSurahSuggestions(value);
+    } else if (this.state.surahs.length > 0) this.setState({ surahs: [] });
+
+    if (value.length >= 3) {
+      this.handleAyahSuggestions(value);
+    } else if (this.state.ayat.length > 0) this.setState({ ayat: [] });
+  };
+
+  handleSurahSuggestions(value) {
+    var escaped = value.replace(/[-\\^$*+?.()|[\]{}]/g, "\\$&");
```
Comment (mmahalwy) [inline src/scripts/components/header/SearchAutocomplete.js]: const

File: src/scripts/components/header/SearchAutocomplete.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    var matches = [];
```
Comment (mmahalwy) [inline src/scripts/components/header/SearchAutocomplete.js]: const

File: src/scripts/components/header/SearchAutocomplete.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    for (var surah_id in this.props.surahs) {
+      var surah = this.props.surahs[surah_id];
```
Comment (mmahalwy) [inline src/scripts/components/header/SearchAutocomplete.js]: const

File: src/scripts/components/header/SearchAutocomplete.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    if (value.length >= 3) {
+      this.handleAyahSuggestions(value);
+    } else if (this.state.ayat.length > 0) this.setState({ ayat: [] });
+  };
+
+  handleSurahSuggestions(value) {
+    var escaped = value.replace(/[-\\^$*+?.()|[\]{}]/g, "\\$&");
+    var matches = [];
+    for (var surah_id in this.props.surahs) {
```
Comment (mmahalwy) [inline src/scripts/components/header/SearchAutocomplete.js]: `suranId`

File: src/scripts/components/header/SearchAutocomplete.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+import { connect } from 'react-redux';
+import ApiClient from '../../../helpers/ApiClient';
+
+@connect(
+  state => ({surahs: state.surahs.entities})
+)
+
+export default class SearchAutocomplete extends Component {
+  constructor(props, context) {
+    super(props, context);
```
Comment (sharabash) [inline src/scripts/components/header/SearchAutocomplete.js]: I don't understand what you mean

File: src/scripts/components/header/SearchAutocomplete.js (hunk 2 above)
Comment (sharabash) [inline src/scripts/components/header/SearchAutocomplete.js]: however `matches` is not a constant, i'm pushing to it

File: src/scripts/components/header/SearchAutocomplete.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import ApiClient from '../../../helpers/ApiClient';
+
+@connect(
+  state => ({surahs: state.surahs.entities})
+)
+
```
Comment (sharabash) [inline src/scripts/components/header/SearchAutocomplete.js]: without this `@connect`, I don't have `this.props.surahs`

**Comment by mmahalwy** [comment]
closing this cause i cannot change your fork

**Pull request closed without merging**